lam4doc/services/handlers.py

```diff
@@ -5,52 +5,99 @@
 # Email: coslet.mihai@gmail.com 
 
-""" """
 import logging
-from json import dumps
 from pathlib import Path
-from shutil import copytree
+from typing import List
+from zipfile import ZipFile
 
 from eds4jinja2.builders.report_builder import ReportBuilder
 
 from lam4doc import config
 from lam4doc.config import LAM_LOGGER
-from lam4doc.entrypoints.api.helpers import generate_report_builder_config
+from lam4doc.services.helpers import FileInfo, IndexInfo
 
 logger = logging.getLogger(LAM_LOGGER)
 
 
-def prepare_report_template(target_location):
+def generate_report(target_location: str, report_builder: ReportBuilder) -> Path:
     """
-    Copy report template and generate the conifg to the temporary directory indicated in target_location
-    :param target_location: path to from where to copy the templates to be used by ReportBuilder
+    Handler for generating the lam report.
+    :param target_location: path to the report templates
+    :param report_builder: a report builder type service (eds4jinja2.builders.report_builder.ReportBuilder)
+    :return: location of the report document
     """
-    logger.debug('start service for preparing lam template ')
-
-    template_location = config.LAM_REPORT_TEMPLATE_LOCATION
-    logger.debug(f'template location {template_location}')
+    logger.debug(f'start service for generating report with target_location: {target_location}')
 
-    copytree(template_location, target_location, dirs_exist_ok=True)
+    report_builder.make_document()
 
-    with open(Path(target_location) / 'config.json', 'w') as config_file:
-        config_content = generate_report_builder_config()
-        config_file.write(dumps(config_content))
+    report_path = Path(str(target_location)) / report_builder.template
 
-    logger.debug('finish service for preparing lam template ')
+    logger.debug(f'finish service for generating report with target_location: {target_location}')
+    return report_path
 
 
-def generate_lam_report(target_location: str, report_builder: ReportBuilder) -> Path:
+def generate_lam_report(target_location: str) -> Path:
     """
-    Handler for generating the lam report.
+    Method for generating the lam report using the generic prepare_template and generate_report
     :param target_location: path to the report templates
-    :param report_builder: a report builder type service (eds4jinja2.builders.report_builder.ReportBuilder)
-    :return: location of the report document
+    :return: path to report
     """
-    logger.debug('start service for generating lam report ')
+    logger.debug('start service for generating lam report')
 
-    report_builder.make_document()
+    report_builder = ReportBuilder(target_path=config.LAM_REPORT_TEMPLATE_LOCATION,
+                                   output_path=target_location,
+                                   additional_config={"conf": {"default_endpoint": config.LAM_FUSEKI_SERVICE}})
+    report_location = generate_report(target_location, report_builder)
```

Comment:
probably we do not need report_generation function anymore.

Reply:
if I remove the method, we lose the dependency injection possibility for the `ReportBuilder`, thus the testing will have to also change

Reply:
I understand now.